plugin/pkg/scheduler/schedulercache/util.go

```diff
@@ -33,3 +36,25 @@ func CreateNodeNameToInfoMap(pods []*api.Pod) map[string]*NodeInfo {
 	}
 	return nodeNameToInfo
 }
+
+type ReferenceWithSelector struct {
+	Reference *api.ObjectReference
+	Selector  labels.Selector
+}
+
+// CreateNodeNameToInfoMapWithGroupingObjects objects calls CreateNodeNameToInfoMap and additionally
+// updates references in NodeInfo structs based on the given pairs of references and selectors.
+func CreateNodeNameToInfoMapWithGroupingObjects(pods []*api.Pod, refs []*ReferenceWithSelector) map[string]*NodeInfo {
```

Comment:
@hongchaodeng - this is purely experimental PR for - please don't spend time on looking into it.
I will let you know if/when it will be ready for review.